src/react-components/tokens/TokensContainer.js

```diff
@@ -0,0 +1,79 @@
+import React, { useState, useEffect, useContext } from "react";
+import { fetchMyTokens } from "./token-utils";
+import { RevokeTokenContainer } from "./RevokeTokenContainer";
+import { TokenList } from "./TokenList";
+import { NoAccess } from "./NoAccess";
+import { AuthContext } from "../auth/AuthContext";
+import { CreateTokenContainer } from "./CreateTokenContainer";
+
+export function TokensContainer() {
+  const [tokens, setTokens] = useState([]);
+  const [isFetching, setIsFetching] = useState(false);
+  const [error, setError] = useState("");
+  const [showCreateToken, setShowCreateToken] = useState(false);
+  const [showRevokeTokenModal, setShowRevokeTokenModal] = useState(false);
+  const [selectedRevokeId, setSelectedRevokeId] = useState();
+
+  const auth = useContext(AuthContext); // Re-render when you log in/out.
+
+  const fetchTokens = async () => {
+    try {
+      setIsFetching(true);
+      setTokens(await fetchMyTokens());
+      setIsFetching(false);
+    } catch (err) {
+      setError("Error fetching tokens: " + err.message);
+      setIsFetching(false);
+      return null;
+    }
+  };
+
+  useEffect(
+    () => {
+      async function updateTokens() {
+        await fetchTokens();
+      }
+      if (auth?.isAdmin) updateTokens();
+    },
+    [auth.isAdmin]
+  );
+
+  const onCreateTokenClose = async ({ createdNewToken }) => {
+    setShowCreateToken(false);
+    if (createdNewToken) await fetchTokens();
+  };
+
+  const onRevokeToken = ({ revokeId }) => {
+    setShowRevokeTokenModal(true);
+    setSelectedRevokeId(revokeId);
+  };
+
+  const onRevokeTokenClose = ({ removedTokenId }) => {
+    if (removedTokenId) setTokens(tokens.filter(token => token.id !== removedTokenId));
+    setShowRevokeTokenModal(false);
+    setSelectedRevokeId("");
+  };
+
+  return (
+    <div>
+      {showRevokeTokenModal && <RevokeTokenContainer tokenId={selectedRevokeId} onClose={onRevokeTokenClose} />}
+      {auth?.isAdmin ? (
```

Comment:
I don't like moving the auth check up one level... It would either need another component to wrap this one or it would be inside TokenPageLayout which wouldn't make sense to me. I'd rather keep the checks in this component so all logic for the tokens pages are centralized in this container component.

It's possible I misunderstood your moving this up one level.